### ISSUE TITLE
error creating requests limiter: iPStrategy and RequestHeaderName are mutually exclusive

### ISSUE THREAD

**Issue by bitgandtter** [containous/traefik#8512]
### Welcome!

- [X] Yes, I've searched similar issues on [GitHub](https://github.com/traefik/traefik/issues) and didn't find any.
- [X] Yes, I've searched similar issues on the [Traefik community forum](https://community.containo.us) and didn't find any.

### What did you do?

create a middleware with

```
apiVersion: traefik.containo.us/v1alpha1
kind: Middleware
metadata:
  name: ratelimit
  namespace: {{ .Values.namespace | quote }}
spec:
  rateLimit:
    average: 1000
    period: 1m
    burst: 1000
    sourceCriterion:
      requestHost: true
      requestHeaderName: X-Real-Ip
      ipStrategy:
        excludedIPs:
          - 127.0.0.1/32
          - 192.168.0.0/24
          - 143.244.220.71
```

### What did you see instead?

error on traefik pods lgos

error creating requests limiter: iPStrategy and RequestHeaderName are mutually exclusive

### What version of Traefik are you using?

2.5.1

### What is your environment & configuration?

default by helm chart

**Comment by tomMoulard** [comment]
Hello @bitgandtter,

Thanks for your interest in Traefik!

As the documentation states:

> The precedence order is `ipStrategy`, then `requestHeaderName`, then `requestHost`.

Which seems to be a documentation error, since the code is quite clear on this:

https://github.com/traefik/traefik/blob/85dd45cb817a017441bdafd92cdaeb360f196ce8/pkg/middlewares/extractor.go#L19-L20

So this issue may be fixed by changing the documentation to match the code.

**Comment by pmareke** [comment]
Hi Tom I would like to help here if there is not problem.

**Comment by traefiker** [comment]
Closed by #8524.
